llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -536,13 +658,26 @@ static bool removeSYCLKernelsConstRefArray(GlobalVariable *GV) {
   return true;
 }
 
-using TableFiles = std::map<StringRef, string_vector>;
+std::unique_ptr<util::SimpleTable>
+processInputModule(std::unique_ptr<Module> M) {
+  // Construct the resulting table which will accumulate all the outputs.
+  SmallVector<StringRef, MAX_COLUMNS_IN_FILE_TABLE> ColumnTitles{
+      StringRef(COL_CODE), StringRef(COL_PROPS)};
+
+  if (DoSymGen) {
+    ColumnTitles.push_back(COL_SYM);
+  }
+  Expected<std::unique_ptr<util::SimpleTable>> TableE =
+      util::SimpleTable::create(ColumnTitles);
+  CHECK_AND_EXIT(TableE.takeError());
+  std::unique_ptr<util::SimpleTable> Table = std::move(TableE.get());
+
+  // Used in output filenames generation.
+  int ID = 0;
 
-TableFiles processOneModule(std::unique_ptr<Module> M, bool IsEsimd,
```

Comment:
this was refactored and in-lined into processInputModule for better clarity:
- table generation was consolidated and simplified
- `CanReuseInputModule` logic also simplified